src/lib/field-directive.ts

```diff
@@ -189,6 +199,7 @@ const createRateLimitDirective = (
         const identityArgs =
           this.args.identityArgs || DEFAULT_FIELD_IDENTITY_ARGS;
         const fieldIdentity = getFieldIdentity(name, identityArgs, resolveArgs);
+        const callCount = this.args.arrayLength ? (resolveArgs[this.args.arrayLength] ? resolveArgs[this.args.arrayLength].length : 1) : 1;
```

Comment:
You may want to consider what to do in the case when the array length is 0. This could be abused.

Reply:
Good catch